warehouse/age-off-utils/src/main/java/datawave/age/off/util/AgeOffRuleConfiguration.java

```
@@ -0,0 +1,148 @@
+            return this;
+        }
+
+        public Builder setWriter(Writer writer) {
+            result.writer = writer;
+            return this;
+        }
+
+        public Builder withIndentation(String indentation) {
+            result.indentation = indentation;
+            return this;
+        }
+
+        public Builder withTtl(String duration, String units) {
+            result.ttlDuration = duration;
+            result.ttlUnits = units;
+            return this;
+        }
+
+        public AgeOffRuleConfiguration build() {
+            if (this.patternConfigurationBuilder != null) {
+                // add two indentations: one for items under the rule element and another for items under the matchPattern element
+                Writer indentingPatternWriter = new IndentingDelegatingWriter(result.indentation + result.indentation, result.getWriter());
+                this.patternConfigurationBuilder.setWriter(indentingPatternWriter);
+                result.patternConfiguration = this.patternConfigurationBuilder.build();
+            }
+            return result;
+        }
+
+        public void addCustomElement(JAXBElement<?> customElement) {
+            if (result.customElements == null) {
+                result.customElements = new ArrayList<>();
+            }
+            result.customElements.add(customElement);
+        }
+    }
+
+    static class IndentingDelegatingWriter extends Writer {
+        private final Writer writer;
+        private final String indentation;
+
+        public IndentingDelegatingWriter(String indentation, Writer writer) {
+            this.indentation = indentation;
+            this.writer = writer;
+        }
+
+        @Override
+        public void write(String str) throws IOException {
+            writer.write(this.indentation + str);
+        }
+
+        @Override
+        public void write(char[] chars, int i, int i1) throws IOException {
+            writer.write(chars, i, i1);
```

For the sake of preserving the API, can we either add the indent here or throw an `UnsupportedOperationException`? That way no one calls this thinking the indent will be respected. I think the arguments are `write(char[] chars, int begin, int length)`

---

Good call.  Addressed in latest push.